apps/computeVSD/computeVSD.cpp

```diff
@@ -0,0 +1,244 @@
+
+/* Copyright (c) 2015-2016, EPFL/Blue Brain Project
+ *                          Jafet.VillafrancaDiaz@epfl.ch
+ *                          Stefan.Eilemann@epfl.ch
+ *                          Daniel.Nachbaur@epfl.ch
+ *
+ * This file is part of Fivox <https://github.com/BlueBrain/Fivox>
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * - Redistributions of source code must retain the above copyright notice, this
+ *   list of conditions and the following disclaimer.
+ * - Redistributions in binary form must reproduce the above copyright notice,
+ *   this list of conditions and the following disclaimer in the documentation
+ *   and/or other materials provided with the distribution.
+ * - Neither the name of Eyescale Software GmbH nor the names of its
+ *   contributors may be used to endorse or promote products derived from this
+ *   software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "beerLambertProjectionImageFilter.h"
+#include "../commandLineApplication.h"
+#include "../volumeHandler.h"
+#include "../volumeWriter.h"
+
+#include <itkImageFileWriter.h>
+
+namespace
+{
+typedef float FloatPixelType;
+typedef itk::Image< FloatPixelType, 2 > FloatImageType;
+}
+
+class ComputeVSD : public CommandLineApplication
+{
+public:
+    ComputeVSD()
+        : CommandLineApplication( "Compute the VSD signal and voxelize it to "
+                                  "generate a 2D image through a Beer-Lambert "
+                                  "projection (VTK format)" )
+        , _outputFile( "image" )
+    {
+        _options.add_options()
+//! [ComputeVSDParameters] @anchor ComputeVSD
+            ( "output,o", po::value< std::string >(),
+              "Name of the output volume file; contains frame number if "
+              "--frames or --times" )
```

Comment:
If the output file format is predefined, mention it here (not in the general description).